### PR TITLE
Python version in config

### PR DESCRIPTION
**Pull request by imphil**
We have many users with Python 2 and Python 3 installed on their machines. The typical setup is to have `python` being Python 2.7, and `python3` being a random Python 3 version. Now confusion comes when people do `pip3 install cocotb` and then use the Makefiles: in this case, PYTHON_BIN is `python`, i.e. the Python 2 version, while cocotb is installed for Python 3. Therefore, the cocotb package won't be found by Python (2), and nothing works. Issue https://github.com/potentialventures/cocotb/issues/944 is an example of that.

This PR fixes this problem by providing the Makefile with the Python version cocotb was installed for. We can leverage the `cocotb-config` tool for that, which is installed as part of the package.

This PR tries hard to keep all the override options we have with `PYTHON_BIN` currently, so users should only get a better experience, but not a worse one. One slight exception is Windows: the PYTHON_DIR environment variable is not used any more. Instead PYTHON_BIN is used like for all other platforms.

### CODE REVIEW

**Comment by imphil** [comment]
Now let's see what CI has to say ...

File: cocotb/share/makefiles/Makefile.pylib.Darwin [hunk 1]
```diff
@@ -29,12 +29,12 @@
 
 # All common python related rules
 
+_PYTHON_BIN?=$(shell cocotb-config --python-bin)
 ifneq ($(COCOTB_PYTHON_DEBUG),)
```
[the remark this answers is not in the document]
Comment (imphil) [reply cocotb/share/makefiles/Makefile.pylib.Darwin]: python-dbg packages are special to (at least) Debian in that they are built with `--with-pydebug`, but use the same site-packages as the non-dbg variant. From the README that Debian installs:

```
Using the python-dbg builds
---------------------------

 * Call the python-dbg or the pythonX.Y-dbg binaries instead of the
   python or pythonX.Y binaries.

 * Properties of the debug build are described in
   /usr/share/doc/@PVER@/SpecialBuilds.txt.gz.
   The debug interpreter is built with Py_DEBUG defined.

 * From SpecialBuilds.txt: This is what is generally meant by "a debug
   build" of Python.  Py_DEBUG implies LLTRACE, Py_REF_DEBUG,
   Py_TRACE_REFS, and PYMALLOC_DEBUG (if WITH_PYMALLOC is enabled).
   In addition, C assert()s are enabled (via the C way: by not defining
   NDEBUG), and some routines do additional sanity checks inside
   "#ifdef Py_DEBUG" blocks.
```

This code was added in 11c504e6 probably as debugging aid (the mentioned issue doesn't talk about the debug flag).

I'm inclined to just remove this functionality since it
- Is not documented.
- Can be easily emulated by setting PYTHON_BIN to the `-dbg` variant.

@stuarthodgson since you added that code in the first place, does that sound OK to you?

**Comment by imphil** [comment]
Note to myself: this needs to go through another iteration, we cannot rely on `cocotb-config` being in the PATH if cocotb is not installed from a package (still a supported configuration).

**Comment by themperek** [comment]
Do we wait with this for 1.3? Will be required to be installed?

PS. We could directly call `config.py` (no install needed) but we need to know python command before.

**Comment by eric-wieser** [comment]
@imphil, did the rebase. Do you want to put this in?

**Comment by imphil** [comment]
I don't think the all edge cases are covered yet, I'll need to test this a bit more. I won't have time until Saturday, unfortunately.

**Comment by imphil** [comment]
Python 2.7 is dead now, no need for this PR any more (thankfully).

**Comment by eric-wieser** [comment]
I found myself wanting this the other day, because `python` on many systems still refers to python 2.

It will become mostly irrelevant after #1306 though.
